### ISSUE TITLE
Can you add src IP address in log?

### ISSUE THREAD

**Issue by XCM-jj** [a2o/snoopy#190]
can you add src ip address in log ?

**Comment by XCM-jj** [comment]
we face a question:
in our company,  some people use only one linux user and  password .
we can not diff them from tty or username.
but we can diff them from src IP address.
like this, use command "w",we can see ip address in the  "FROM" :
```
[sss ~]$w
 19:25:52 up 2 days,  5:17,  3 users,  load average: 0.01, 0.02, 0.05
USER     TTY      **FROM**             LOGIN@   IDLE   JCPU   PCPU WHAT
w      pts/1    10.3.0.34      17:35    3:04   0.08s  0.02s sshd: w [priv]    
w      pts/2    10.3.0.34      17:37    0.00s  0.02s  0.00s w
```


thank you .

**Comment by bostjan** [comment]
If your users are connecting to this host via SSH (which is most likely the case), you can get IP address information from the environment variable right now, provided that child processes do not remove it explicitly. You can do so by adding `%{env:SSH_CLIENT}` to your `message_format` configuration variable in your `snoopy.ini`.

That said, `w`/`who` tools get IP address information from the `utmp` database, and implementing support for that will not happen in the short term. It's a good suggestion, though, so I'll leave this ticket open.

**Comment by XCM-jj** [comment]
thank you .

**Comment by XCM-jj** [comment]
> 
> 
> If your users are connecting to this host via SSH (which is most likely the case), you can get IP address information from the environment variable right now, provided that child processes do not remove it explicitly. You can do so by adding `%{env:SSH_CLIENT}` to your `message_format` configuration variable in your `snoopy.ini`.
> 
> That said, `w`/`who` tools get IP address information from the `utmp` database, and implementing support for that will not happen in the short term. It's a good suggestion, though, so I'll leave this ticket open.

hello ， I want to tell you to attention:
i find i can get IP address in` env:SSH_CLIENT` by the time i log in .
But ,by the time, i use `sudo -s`  to let the user become root, the `env` do not have the key word `SSH_CLIENT` any more.

**Comment by bostjan** [comment]
@XCM-jj, you _can_ configure what environmental variables sudo retains.

Here is one example resource documenting the behavior:
https://superuser.com/questions/232231/how-do-i-make-sudo-preserve-my-environment-variables

**Comment by XCM-jj** [comment]
@bostjan   I left tears of no technology.

**Comment by bostjan** [comment]
@XCM-jj, I have no idea what your statement `I left tears of no technology` is supposed to mean, sorry. :)

**Comment by XCM-jj** [comment]
@bostjan 
`I left tears of no technology`，this is a Chinese programmer jokes.

**Comment by bostjan** [comment]
Implemented via #233.

It can be enabled with the `ipaddr` data source in `message_format` setting:
```ini
[snoopy]
message_format = ip:%{ipaddr} %{cmdline}`
```

This feature will be released in v2.5.0.

@XCM-jj thanks for suggesting this.
